crates/dyn-abi/src/ty.rs

```diff
@@ -147,7 +165,7 @@ impl DynSolType {
     /// ```
     #[inline]
     pub fn parse(s: &str) -> DynAbiResult<Self> {
-        TypeSpecifier::try_from(s).and_then(|t| t.resolve_basic_solidity())
+        TypeSpecifier::parse(s)?.resolve()
```

Comment:
`and_then` has better codegen, I think re-uses the first error's stack allocation or something

Reply:
`and_then` cannot be used to change the result type. I'll remove the `?` another way

Reply:
If they have different error then that's fine

Reply:
used a map_err, which should still have better codegen than `?`